### DISCUSSION TITLE
Missing option to disallow federation on public spaces

### DISCUSSION THREAD

**Discussion by airblag** [element-hq/element-meta#741] [Ideas]
#### What would you like to do?

My element on my homeserver is configured with a config.json specifiying `"default_federate": false`
This makes my user create rooms which are per default not federated. This sadly didn't apply to the newly created spaces.

I want the same pre-filled option when creating spaces that is shown in the advanced options when creating a room.

#### Why would you like to do it?

In an enterprise environment, it's useful to have "public spaces" that all internal users can join by following a link. The spaces also all have an #alias:homeserver.net short name, which makes them quite easy to guess for an attacker.

The rooms inside the space might not be joinable (because mostly created with the non federable flag), but this might leak the rooms and users of the space, which is not acceptable from a security perspective.

#### How would you like to achieve it?

I would love to see the default_federate option of the config.json being applied to newly created spaces. This would also mean to change the UI for the spaces creation to include the option for enabling federation.

Right now, all the public spaces on my homeserver are joinable by external users by default, and it's not easy to change (I need to apply ACL to all spaces already created). The alternative is to mark all the spaces as private, but this is breaking part of the workflow in the company, since we'll have to invite every user individually to all the possible spaces.

Since the federation flag can only be set on creation of the room/space, it would also be a good thing to have a UI to apply the ACL for denying federated access.

There was already a bug report in this direction on the synapse repo, but this as to be fixed in the client.
https://github.com/matrix-org/synapse/issues/11227

**Comment by ZelnickB** [comment]
Is there any timeline for when this feature is coming? At this point, I'm using the `curl` command at <https://github.com/matrix-org/synapse/issues/11227#issuecomment-963079347> to achieve this, and it is quite inconvenient. Since a space is a type of room, it doesn't make sense for the `default_federate` option *not* to be applied to newly-created spaces as well, and it might be reasonable to consider this a software bug.
